### PR TITLE
[Merged by Bors] - lifetime related cleanup in `entity_ref.rs`

### PR DESCRIPTION
**Pull request by BoxyUwU**
# Objective

EntityMut::world takes &mut self instead of &self I don't see any reason for this.
EntityRef is overly restrictive with fn world and could return &'w World

---

## Changelog

- EntityRef now implements Copy and Clone
- EntityRef::world is now fn(&self) -> &'w World instead of fn(&mut self) -> &World
- EntityMut::world is now fn(&self) -> &World instead of fn(&mut self) -> &World

### CODE REVIEW

File: crates/bevy_ecs/src/world/entity_ref.rs [hunk 1]
```diff
@@ -44,7 +45,7 @@ impl<'w> EntityRef<'w> {
     }
 
     #[inline]
-    pub fn world(&mut self) -> &World {
+    pub fn world(self) -> &'w World {
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/world/entity_ref.rs:48]: Why is the lifetime explicit here but not below in the `EntityMut` equivalent?

Comment (alice-i-cecile) [inline crates/bevy_ecs/src/world/entity_ref.rs:48]: Why does this take a `self` rather than `&self`?

Comment (BoxyUwU) [reply crates/bevy_ecs/src/world/entity_ref.rs:48]: since its `Copy` theres no real reason to take `&self` It doesn't really make a difference

Comment (BoxyUwU) [reply crates/bevy_ecs/src/world/entity_ref.rs:48]: `EntityMut` ties a borrow to `&self` wheras this doesnt

Comment (cart) [reply crates/bevy_ecs/src/world/entity_ref.rs:48]: Seems a bit weird to consume / take ownership here, even if Copy makes this behave normally. Unless we have codegen reasons to use this pattern, I think it creates fewer questions in the reader's mind if we use `&self`. And using copying here also involves "more" data transfer than passing in a `self` pointer. Probably doesn't matter, but that feels like another reason to bias toward `&self`.

**Review by alice-i-cecile — APPROVED** [review]

**Review by james7132 — APPROVED** [review]

**Pull request closed without merging**